lib/usd/utils/MayaTransformAPI.cpp

```diff
@@ -0,0 +1,1046 @@
+        if(++iter == end)
+          return true;
+        type = iter->GetOpType();
+        name = iter->GetOpName();
+      }
+    }
+    break;
+
+  default:
+    break;
+  }
+
+  if(tokens::rotateAxis == name)
+  {
+    m_rotateAxis = *iter;
+    if(++iter == end)
+      return true;
+    type = iter->GetOpType();
+    name = iter->GetOpName();
+  }
+
+  if(type == UsdGeomXformOp::TypeTranslate && XFORM_NAME_CHECK(tokens::rotatePivotINV))
+  {
+    m_rotatePivotINV = *iter;
+    if(++iter == end)
+      return m_api != TransformAPI::kCommon;
+    type = iter->GetOpType();
+    name = iter->GetOpName();
+  }
+
+  if(type == UsdGeomXformOp::TypeTranslate && name == tokens::scalePivotTranslate)
+  {
+    m_scalePivotTranslate = *iter;
+    if(++iter == end)
+      return m_api != TransformAPI::kCommon;
+    type = iter->GetOpType();
+    name = iter->GetOpName();
+  }
+
+  if(type == UsdGeomXformOp::TypeTranslate && name == tokens::scalePivot)
+  {
+    m_scalePivot = *iter;
+    if(++iter == end)
+      return m_api != TransformAPI::kCommon;
+    type = iter->GetOpType();
+    name = iter->GetOpName();
+  }
+
+  if(type == UsdGeomXformOp::TypeTransform && name == tokens::shear)
+  {
+    m_shear = *iter;
+    if(++iter == end)
+      return true;
+    type = iter->GetOpType();
+    name = iter->GetOpName();
+  }
+
+  if(type == UsdGeomXformOp::TypeScale && XFORM_NAME_CHECK(tokens::scale))
+  {
+    m_scale = *iter;
+    if(++iter == end)
+      return true;
+    type = iter->GetOpType();
+    name = iter->GetOpName();
+  }
+
+  if(type == UsdGeomXformOp::TypeTranslate && XFORM_NAME_CHECK(tokens::scalePivotINV))
+  {
+    m_scalePivotINV = *iter;
+    if(++iter == end)
+      return m_api != TransformAPI::kCommon;
+    type = iter->GetOpType();
+    name = iter->GetOpName();
+  }
+
+  if(type == UsdGeomXformOp::TypeTranslate && name == tokens::pivotINV)
+  {
+    m_pivotINV = *iter;
+    if(++iter == end)
+      return m_api == TransformAPI::kCommon;
+  }
+
+  return false;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+MayaTransformAPI::MayaTransformAPI(UsdPrim prim, bool convertMatrixOpToComponentOps)
+  : m_prim()
+{
+  bool reset = false;
+  std::vector<UsdGeomXformOp> ops = UsdGeomXformable(prim).GetOrderedXformOps(&reset);
+  if(matchesMayaTrasformProfile(ops))
+  {
+    // only assign if we can process the data
+    m_prim = prim;
+  }
+
+  // if the prim is not valid, check to see if we have a transform Op.
+  if(convertMatrixOpToComponentOps && !m_prim)
+  {
+    if(ops.size() == 1 && ops[0].GetOpType() == UsdGeomXformOp::TypeTransform)
+    {
+      m_prim = prim;
+      this->convertMatrixOpToComponentOps(ops[0]);
+    }
+  }
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void MayaTransformAPI::inheritsTransform(const bool inherit)
+{
+  if(m_prim)
```

Comment:
In some sense you've chosen silent failure for invalid prims (here and elsewhere in your implementation).  Have you thought about raising an exception?